### PR TITLE
Initial Shield Overspill Rework

### PR DESCRIPTION
**Pull request by Azraeel**

### CODE REVIEW

**Comment by Azraeel** [comment]
@loudpatches I'd like you to do a in-depth debug session on my work here to ensure it's all working, I did not seeing any obviously wrong with my code etc but it seems like the overspill is not applying.

File: hook/lua/shield.lua [hunk 1]
```diff
@@ -0,0 +1,944 @@
...
+
+            -- retrieve shields that overlap with us
+            local others, count = self:GetOverlappingShields(tick)
+
+            LOG("What is the Overlapping Shield Amount (others) "..repr(others).. "and Count is "..repr(count))
+
+            -- apply overspill damage to neighbour shields
+            for k = 1, count do
+                others[k]:ApplyDamage(
+                    LOG("Applying Overspill Damage to other shields nearby"),
```
Comment (rajderks) [inline hook/lua/shield.lua:546]: This log might be the culprit

**Review by rajderks — CHANGES_REQUESTED** [review]
Found 1 possible error

File: hook/lua/shield.lua (hunk 1 above)
Comment (Azraeel) [reply hook/lua/shield.lua]: Should be fixed now, still working on figuring out why the overspill isnt applying to shields.

**Pull request merged**